### PR TITLE
RDoc-2501_and_2522_breaking

### PR DESCRIPTION
**Pull request by reebhub**
Breaking changes, licensing section updates, added casting warnings to sharding pages,

### CODE REVIEW

File: Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  is now `true`.  
+
+* **Facets**  
+ `FacetOptions` were removed from `RangeFacet`.  
+
+* **Obsolete entities removed**  
+  Many obsolete attributes, properties, and methods were removed from the public API.  
+
+* A `DisposeCertificate` convention has been added to prevent/allow the disposal of 
+  `DocumentStore.Certificate` when the store is disposed of.
```
Comment (ppekrol) [inline Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown:47]: at the end add: `to help users mitigate the X509Certificate2 leak. More info [here](https://snede.net/the-most-dangerous-constructor-in-net/)`

File: Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+ `FacetOptions` were removed from `RangeFacet`.  
+
+* **Obsolete entities removed**  
+  Many obsolete attributes, properties, and methods were removed from the public API.  
+
+* A `DisposeCertificate` convention has been added to prevent/allow the disposal of 
+  `DocumentStore.Certificate` when the store is disposed of.  
+
+* **Serialization**  
+  Only **Public** fields are serialized/deserialized.
```
Comment (ppekrol) [inline Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown:50]: when doing a projection

File: Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+   * [Breaking Changes](../../migration/client-api/client-breaking-changes#breaking-changes)  
+   * [Breaking Changes in a Sharded Database](../../migration/client-api/client-breaking-changes#breaking-changes-in-a-sharded-database)  
+   * [Additional breaking changes](../../migration/client-api/client-breaking-changes#additional-breaking-changes)  
+
+{NOTE/}
+
+{PANEL: Breaking Changes}
+
+* **ETL**  
+  SQL ETL now tolerates errors on `Load`: load errors are thrown immediately, to distinguish
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown:19]: done

File: Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  registered in it, to indicate that the operation is forbidden from a non-tracking session 
+  and warn about its expected results.  
+  `"This session does not track any entities, because of that registering includes is forbidden  
+  to avoid false expectations when later load operations are performed on those and no requests 
+  are being sent to the server. Please avoid any 'Include' operations during non-tracking session 
+  actions like load or query."`
+
+* **Type Changes**  
+  Many methods and types that used the `int` type in former RavenDB versions now 
+  use `long`, e.g. `BlittableArrayResult.TotalResults` and `QueryStatistics.TotalResults`.
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown:34]: done

File: Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  to avoid false expectations when later load operations are performed on those and no requests 
+  are being sent to the server. Please avoid any 'Include' operations during non-tracking session 
+  actions like load or query."`
+
+* **Type Changes**  
+  Many methods and types that used the `int` type in former RavenDB versions now 
+  use `long`, e.g. `BlittableArrayResult.TotalResults` and `QueryStatistics.TotalResults`.  
+
+* **Indexing**  
+  The default value of [Indexing.OrderByTicksAutomaticallyWhenDatesAreInvolved](../../server/configuration/indexing-configuration#indexing.orderbyticksautomaticallywhendatesareinvolved)
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown:37]: done

File: Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown:47]: done

File: Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown (hunk 2 above)
Comment (reebhub) [reply Documentation/6.0/Raven.Documentation.Pages/migration/client-api/client-breaking-changes.markdown:50]: done

**Pull request merged**